src/workflow/Files.py

```diff
@@ -0,0 +1,215 @@
+from pathlib import Path
+import string
+import random
+import shutil
+import streamlit as st
+from typing import Union, List
+from .Logger import Logger
+
+
+class Files:
+    """
+    Manages file paths for operations such as changing file extensions, organizing files
+    into result directories, and handling file collections for processing tools. Designed
+    to be flexible for handling both individual files and lists of files, with integration
+    into a Streamlit workflow.
+
+    Attributes:
+        files (List[str]): A list of file paths, initialized from various input formats.
+        file_type (str, optional): The default file type/extension for the files.
+        results_dir (str, optional): The directory to store processed results.
+    """
+    def __init__(
```

Comment:
these classes have some state. it is hard to see for me (without digging into other parts of the code) if this objects exists once per instance, once per session, etc. 
it is also not clear why global variables like the streamlit object needs to be accessed? if not I would consider constructing the Files object with the workspace folder. Then it is much clearer that one object is bound to one particular workflow folder